src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb

```diff
@@ -661,8 +665,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return DirectCast(node, ParenthesizedExpressionSyntax).Expression
         End Function
 
-        Public Function IsIfStatement(node As SyntaxNode) As Boolean Implements ISyntaxFactsService.IsIfStatement
-            Return (node.Kind() = SyntaxKind.IfStatement)
+        Public Function GetIfStatementCondition(node As SyntaxNode) As SyntaxNode Implements ISyntaxFactsService.GetIfStatementCondition
+            If TypeOf node Is MultiLineIfBlockSyntax Then
+                Return DirectCast(node, MultiLineIfBlockSyntax).IfStatement.Condition
+            ElseIf TypeOf node Is ElseIfBlockSyntax Then
+                Return DirectCast(node, ElseIfBlockSyntax).ElseIfStatement.Condition
+            End If
```

Comment:
te issue here is that this is codifying an ElseIfBlock as an 'IfStatement' from the SyntaxFacts view of things.  But it's unclear to me that that's something we'd want to do here.  For example, if ISyntaxFacts has IsIfStaement, people may think it should *only* be for 'IfBlocks'.

In general, ISyntaxFacts 'If' queries are to blur the lines over equivalent C#/VB features.  They are *not* to blur hte lines over similar C#/C# or VB/VB features.  People have added stuff like that on occasion, and it's been highly problematic (for example, helpers to try to make it seem as if `a.b` is the same as `a?.b` and whatnot).

Reply:
I understand. The idea was that `ElseIf` is really similar to C# `else` and an `if`, just in one node.

Reply:
Which would you prefer then:
1. just renaming the syntax facts method to use IfLikeStatement
2. having separate syntax facts for if and else if, and having the feature check both
3. removing any of this from syntax facts altogether and add feature-specific language-specific helpers?

Reply:
➡️ Resolved by moving this out of `ISyntaxFactsService` into a new feature specific service, `IIfLikeStatementGenerator`.